src/components/NewDeckForm/NewDeckForm.test.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import * as rtl from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import NewDeckForm from "./NewDeckForm";
+
+const mockDispatch = jest.fn();
+jest.mock("react-redux", () => ({
+  useSelector: jest.fn(),
+  useDispatch: () => mockDispatch,
+}));
+
+// cleaning up
+afterEach(() => {
+  rtl.cleanup();
+});
+
+let wrapper;
+
+beforeEach(() => {
+  wrapper = rtl.render(<NewDeckForm />);
+});
+
+// useHistory mock so tests dont crash
+jest.mock("react-router-dom", () => ({
+  useHistory: () => ({
+    push: jest.fn(),
+    location: {
+      pathname: "random/text/okay/token",
+    },
+  }),
+  Link: "a",
+}));
+
+jest.mock("react-redux", () => ({
+  useDispatch: jest.fn(),
+}));
```

Comment:
You are mocking 'react-redux' 2 times here,

Reply:
Fixed